### PR TITLE
CONTRIBUTING.md: migrate parts of wiki and add general tips

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This moves some of the contributing guidelines from various wiki pages to `CONTRIBUTING.md` in an effort to centralize documentation.

It also adds a list of tips, that we discussed offline today.

### Testing procedure
Read!

### Issues/PRs references
Addresses #12779.

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
I looked over this and recommended what I could, maybe @kaspar030, @haukepetersen and @smlng could give some input.

**Comment by MrKevinWeiss** [comment]
and @OlegHahm

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,6 +20,103 @@ As a reminder, all contributors are expected to follow our
...
+#### Code comments, HOWTOs for particular projects
+Documentation that relates directly to the code at hand like the HOWTO files
+that can be found in some of the directories in ``RIOT/examples/`` or comments
+in the code itself should be submitted through a [pull request][pull-requests].
+
+If you're not sure about the correct way to submit your writing, please ask on
+the mailing list or open an issue saying which documentation is missing. The
+other RIOTers will help you find the right format.
+
+### Issues
```
Comment (miri64) [inline CONTRIBUTING.md]: Please note that I think that this subsection ("Issues") and the subsection that follows ("Contribute code") somewhat duplicate what is already described in "Feature Requests", "Bug Reports", and "Pull Requests", just in a more concise form

**Comment by miri64** [comment]
Maybe also @aabadie? He wrote most of the original text in `CONTRIBUTING.md`

**Comment by miri64** [comment]
Ping?

**Comment by miri64** [comment]
Ping?

**Comment by fjmolinas** [comment]
ACK! Please squash.

**Comment by miri64** [comment]
> ACK! Please squash.

Done

**Pull request merged**